plugins/ltac2/tac2quote.ml

```diff
@@ -123,6 +130,26 @@ let of_preterm ?delimiters c =
   let c = quote_constr ?delimiters c in
   inj_wit ?loc wit_preterm c
 
+let of_open_constr_expected_istype ?delimiters c =
```

Comment:
It may be easier to add a `open_constr_istype:()` CTacExt but I'm not sure that we want that kind of proliferation.
OTOH this produces slightly different results from intern_type when a scope is bound to Sortclass but does not have delimiters (intern_type would use the scope but this code doesn't).

It's pretty similar to `open_constr:(c :> Type)` but also works for SProp, doesn't force to refresh universes for algebraics etc. 
Maybe we could have some syntax that says "this term is a type"?
IDK